apiserver/facades/client/leadership/leadership.go

```diff
@@ -69,15 +69,20 @@ func (a *api) UnpinLeadership(param params.PinLeadershipBulkParams) (params.Erro
 	return a.pinOps(param, a.pinner.UnpinLeadership), nil
 }
 
-func (a *api) pinOps(param params.PinLeadershipBulkParams, op func(string) error) params.ErrorResults {
+func (a *api) pinOps(param params.PinLeadershipBulkParams, op func(string, names.Tag) error) params.ErrorResults {
 	results := make([]params.ErrorResult, len(param.Params))
 	for i, p := range param.Params {
-		tag, err := names.ParseApplicationTag(p.ApplicationTag)
+		appTag, err := names.ParseApplicationTag(p.ApplicationTag)
 		if err != nil {
 			results[i] = params.ErrorResult{Error: common.ServerError(err)}
 			continue
 		}
-		if err = op(tag.Id()); err != nil {
+		entityTag, err := names.ParseTag(p.EntityTag)
+		if err != nil {
+			results[i] = params.ErrorResult{Error: common.ServerError(err)}
+			continue
+		}
```

Comment:
This is the part where we should be doing an Authorization check to make sure that the entity making the request has the right to mark it on behalf of the entity. (machine-0 shouldn't be able to request a pin for machine-1)